### PR TITLE
feat: Add button to force reinstall Northstar

### LINKED ISSUE #153: Add option for force reinstall Northstar
Add some button in the repair or dev menu that allows for force-reinstalling Northstar.

Mostly need this for myself to go back to latest release after installing some dev version.

### PR DESCRIPTION
**Pull request by GeckoEidechse**
Useful for reinstalling Northstar after installing a developer version.

Closes #153

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Note that it doesn't update the version number displayed on the main / PlayView and I kinda don't know how to get it to update ^^"

File: src-vue/src/views/DeveloperView.vue [hunk 1]
```diff
@@ -207,7 +212,22 @@ export default defineComponent({
...
+        },
+        async forceInstallNorthstar() {
+            let game_install = {
+                game_path: this.$store.state.game_path,
+                install_type: this.$store.state.install_type
+            } as GameInstall;
+            let install_northstar_result = invoke("install_northstar_caller", { gamePath: game_install.game_path, northstarPackageName: ReleaseCanal.RELEASE });
+            await install_northstar_result
+                .then((message) => {
+                    console.log(message);
```
Comment (Alystrasz) [inline src-vue/src/views/DeveloperView.vue:224]: Is this needed?

Comment (Alystrasz) [inline src-vue/src/views/DeveloperView.vue:224]: So you should update installed version here (the variable to update is `installed_northstar_version`)

**Review by Alystrasz — CHANGES_REQUESTED** [review]

File: src-vue/src/views/DeveloperView.vue (hunk 1 above)
Comment (GeckoEidechse) [reply src-vue/src/views/DeveloperView.vue:224]: Wouldn't the best approach be to just call `_get_northstar_version_number` from `store.ts`?

https://github.com/R2NorthstarTools/FlightCore/blob/d75ba3fe4b4b95b50490e0d8febdaf7d5a4c75ce/src-vue/src/plugins/store.ts#L453-L474

But as far as I can tell that's a private function?

Comment (GeckoEidechse) [reply src-vue/src/views/DeveloperView.vue]: No, removed in 1ffd1c3bb404db3c69b6242b2bc25b9051fd0afb (and replaced with `TODO` comment for https://github.com/R2NorthstarTools/FlightCore/pull/154#issuecomment-1412881556 :P)

Comment (Alystrasz) [reply src-vue/src/views/DeveloperView.vue]: Yep, but you still can invoke it through the `checkNorthstarUpdates` store mutation:
```javascript
store.commit('checkNorthstarUpdates');
```

Comment (GeckoEidechse) [reply src-vue/src/views/DeveloperView.vue]: Aight, fixed in ed315e25b2d9c5216f272b44224fc9c140332fdc

**Review by Alystrasz — CHANGES_REQUESTED** [review]
Looks good and works fine on Windows.

Could you add UI feedback to acknowledge FlightCore is currently reinstalling Northstar?
Currently, when you click the button, nothing happens on the interface.

**Comment by GeckoEidechse** [comment]
> Could you add UI feedback to acknowledge FlightCore is currently reinstalling Northstar?
> Currently, when you click the button, nothing happens on the interface.

Any idea how I can implement that in a quick way in a few lines of code? ^^
Frontend is definitely not my forté and I rather not have the PR be blocked by me trying to figure out frontend stuff for a week :P

At the very least I could show a notification when it's done if that's enough.
(I usually just look at the console output cause I'm running from a local build :P)

**Comment by Alystrasz** [comment]
> At the very least I could show a notification when it's done if that's enough.

Given that's a developer feature, a notification when starting downloading and a notification when it's done should do the trick :)

**Comment by GeckoEidechse** [comment]
> > At the very least I could show a notification when it's done if that's enough.
> 
> Given that's a developer feature, a notification when starting downloading and a notification when it's done should do the trick :)

Done as suggested in af881ad45698cac236d10749d0fbd7e7ca9e2090 ^^

**Review by Alystrasz — APPROVED** [review]
LGTM.
Confirmed working on Windows.

**Pull request merged**